lib/ui/router.dart

```diff
@@ -1,52 +1,13 @@
-import 'package:flutter/cupertino.dart';
-import 'package:flutter/material.dart';
-import 'package:get/get.dart';
-import 'package:provider_start/core/constant/view_routes.dart';
-import 'package:provider_start/core/models/post/post.dart';
+import 'package:auto_route/auto_route_annotations.dart';
 import 'package:provider_start/ui/views/login/login_view.dart';
 import 'package:provider_start/ui/views/main/main_view.dart';
 import 'package:provider_start/ui/views/startup/start_up_view.dart';
 import 'package:provider_start/ui/widgets/stateful/post_details/post_details_view.dart';
 
-/// Class that generates routes for the application
-///   - Routes are generated by the list of routes in the app
-///   - Routes can also require parameters. e.g. `PostDetailView(post: post)`
-class Router {
-  static Route<dynamic> generateRoute(RouteSettings settings) {
-    return GetPageRoute(
-      settings: RouteSettings(name: settings.name),
-      page: () => _generateView(settings),
-      fullscreenDialog: _fullScreenDialogs.contains(settings.name),
-    );
-  }
-
-  static Widget _generateView(RouteSettings settings) {
-    switch (settings.name) {
-
-      // Tab Views
-      case ViewRoutes.main:
-        return MainView();
-      case ViewRoutes.login:
-        return LoginView();
-
-      case ViewRoutes.splash:
-        return StartUpView();
-      case ViewRoutes.post_details:
-        final post = settings.arguments as Post;
-        return PostDetailsView(post: post);
-
-      default:
-        return Scaffold(
-          body: Center(
-            child: Text('No route defined for ${settings.name}'),
-          ),
-        );
-    }
-  }
-
-  // Add routes that should behave as fullScreenDialogs
-  static final _fullScreenDialogs = [
-    // Routes.route_1,
-    // Routes.route_2,
-  ];
-}
+@MaterialAutoRouter(routes: [
```

Comment:
Please use adaptive routes for material and Cupertino routes

Reply:
Will fix